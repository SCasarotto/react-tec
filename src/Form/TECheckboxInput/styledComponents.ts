import styled from 'styled-components'

export const Input = styled.input`
	cursor: pointer;
	box-sizing: border-box;
	margin: 0px; /* Resets Safari Margin */

	:disabled {
		cursor: not-allowed;
	}
`
