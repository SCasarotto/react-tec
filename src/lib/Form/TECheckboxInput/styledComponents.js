import styled from 'styled-components'

export const Input = styled.input`
	${(props) => {
		return `
			cursor: pointer;
            box-sizing: border-box;
		`
	}}
`
