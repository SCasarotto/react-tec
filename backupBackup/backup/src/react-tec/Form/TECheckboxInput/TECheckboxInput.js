//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import { Input } from './styledComponents'

const TECheckboxInput = (props) => {
    return <Input {...props} type="checkbox" />
}

TECheckboxInput.propTypes = {
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
    checked: PropTypes.bool,
    id: PropTypes.string,
}

export default TECheckboxInput
