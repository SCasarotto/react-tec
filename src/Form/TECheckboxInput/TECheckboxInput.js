//
//TE Version 0.3.0
//

import React from 'react'

import { Input } from './styledComponents'

const TECheckboxInput = (props) => {
	return <Input {...props} type="checkbox" />
}

export default TECheckboxInput
