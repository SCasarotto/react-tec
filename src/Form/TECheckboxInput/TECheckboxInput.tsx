import React from 'react'

import { Input } from './styledComponents'

export interface TECheckboxInputProps extends React.InputHTMLAttributes<HTMLInputElement> {}
export const TECheckboxInput = React.forwardRef<HTMLInputElement, TECheckboxInputProps>(
	(props, ref) => {
		const { className = '', ...rest } = props
		return (
			<Input className={`TECheckboxInput ${className}`} {...rest} ref={ref} type='checkbox' />
		)
	},
)
