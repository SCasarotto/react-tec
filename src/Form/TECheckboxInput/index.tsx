import React from 'react'

import { Input } from './styledComponents'

export interface TECheckboxInputProps
	extends React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement> {
	//TODO: Fix this. styled-components is weird with refs
	ref?: any
}
export const TECheckboxInput: React.FC<TECheckboxInputProps> = (props) => {
	const { className = '', ...rest } = props
	return <Input className={`TECheckboxInput ${className}`} {...rest} type='checkbox' />
}
