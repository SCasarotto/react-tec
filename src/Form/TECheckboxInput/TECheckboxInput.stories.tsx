import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TECheckboxInput, TECheckboxInputProps } from './TECheckboxInput'

export default {
	title: 'Form/TECheckboxInput',
	component: TECheckboxInput,
	argTypes: {},
} as Meta

const Template: Story<TECheckboxInputProps> = (args) => <TECheckboxInput {...args} />

export const Default = Template.bind({})
Default.args = {
	value: 'soemValue',
}
