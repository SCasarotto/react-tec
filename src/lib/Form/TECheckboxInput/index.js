import TECheckboxInput from './TECheckboxInput'

export default TECheckboxInput
