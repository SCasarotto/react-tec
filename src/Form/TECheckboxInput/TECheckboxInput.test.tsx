import React from 'react'
import { render } from '@testing-library/react'
import { TECheckboxInput } from './TECheckboxInput'

describe('TECheckboxInput', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(<TECheckboxInput />)
		expect(asFragment()).toMatchSnapshot()
	})
})
