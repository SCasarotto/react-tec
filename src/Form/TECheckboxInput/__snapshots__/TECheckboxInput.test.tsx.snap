// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TECheckboxInput matches snapshot 1`] = `
<DocumentFragment>
  <input
    class="sc-bdnylx dpqFiO TECheckboxInput "
    type="checkbox"
  />
</DocumentFragment>
`;
